### DISCUSSION TITLE
关于webdav存储策略的设置问题

### DISCUSSION THREAD

**Discussion by Neo-peng** [lsky-org/lsky-pro#417] [问答]
我想要上传到webdav中，下面是我的webdav存储策略的配置
[![1652165935709.png](https://imgs.mcurr.xyz/i/2022/05/10/627a0d315ac85.png)](https://imgs.mcurr.xyz/i/2022/05/10/627a0d315ac85.png)

但是上传到这个webdav策略的时候，提示如下错误: "上传失败, 服务异常，请稍后再试"
[![1652168077744.png](https://imgs.mcurr.xyz/i/2022/05/10/627a158e25d70.png)](https://imgs.mcurr.xyz/i/2022/05/10/627a158e25d70.png)

查看了上传的错误日志，部分日志如图：
![](https://imgs.mcurr.xyz/i/2022/05/10/627a15fe8cc91.png)

因为没有使用过php，所以不清楚问题出在。检查过我的webdav服务器是正常的，因为用其他的客户端软件可以正常连接webdav，所以猜测应该是webdav存储策略出现了错误，但是不清楚问题在哪里，尝试过域名后面加上和不加上 '/dav' 的uri，但是都上传失败，如果有使用webdav存储策略的成功的，还请不吝赐教。

**Comment by penndu** [comment]
方便留一个联系方式吗？

**Comment by 0xxb** [comment]
目测应该是 curl 拓展问题，重新编译安装一下 curl 拓展试试

**Reply by Neo-peng** [reply]
装了curl拓展后，确实解决了上面的问题，已经可以给webdav服务器发送API请求，但是又出现了下面这个错误，这个是不是和跨域有关，因为我没有在服务器装宝塔，是手动装的lsky，所以看教程安装时开启cors的步骤我给跳了过去。
上传错误日志截图：
![1652264980649.png](https://imgs.mcurr.xyz/i/2022/05/11/627b901696dce.png)

**Reply by penndu** [reply]
> 装了curl拓展后，确实解决了上面的问题，已经可以给webdav服务器发送API请求，但是又出现了下面这个错误，这个是不是和跨域有关，因为我没有在服务器装宝塔，是手动装的lsky，所以看教程安装时开启cors的步骤我给跳了过去。 上传错误日志截图： ![1652264980649.png](https://camo.githubusercontent.com/1fbf7ccbe4d46e181280db70d598ff20720b65922e755730ef85b13fb0b2c103/68747470733a2f2f696d67732e6d637572722e78797a2f692f323032322f30352f31312f363237623930313639366463652e706e67)

![image](https://user-images.githubusercontent.com/46226528/167829904-c8d10ee6-5403-4936-81f9-287b92d40025.png)

**Reply by Neo-peng** [reply]
这个是什么，我的问题出在没法通过lskypro的webdav策略上传
